src/Tribe/Context.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+/**
+ * Class Tribe__Context
+ *
+ * @since TBD
+ */
+class Tribe__Context {
+
+	/**
+	 * Whether we are currently creating a new post, a post of post type(s) or not.
+	 *
+	 * @since TBD
+	 *
+	 * @param null|array|string|int $post_or_type A post type, an array of post types, `null`
+	 *                                            to just make sure we are currently creating a post.
+	 *
+	 * @return bool
+	 */
+	public function is_new_post( $post_type = null ) {
+		global $pagenow;
+		$is_new = $pagenow === 'post-new.php';
+
+		return $is_new && $this->is_editing_post( $post_type );
+	}
+
+	/**
+	 * Whether we are currently editing a post(s), post type(s) or not.
+	 *
+	 * @since TBD
+	 *
+	 * @param null|array|string|int $post_or_type A post ID, post type, an array of post types or post IDs, `null`
+	 *                                            to just make sure we are currently editing a post.
+	 *
+	 * @return bool
+	 */
+	public function is_editing_post( $post_or_type = null ) {
+		global $pagenow;
+		$is_new  = $pagenow === 'post-new.php';
+		$is_post = $pagenow === 'post.php';
+
+		if ( ! ( $is_new || $is_post ) ) {
```

Comment:
🤔 maybe this is just a matter of personal preference, but for me it would read more easily if we switched this to `if ( ! $is_new && ! $is_post )` - minimizes bracketing and - to me - reads clearly as _if this is not new and is not a post, do this next thing._ 

No need to change though if `( ! ( $is_new || $is_post ) )` is clearer to you.